GetServiceByLatestRevision/handler.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Implements the Public API handlers for the Services resource.
+ */
+
+import * as express from "express";
+
+import {
+  IResponseErrorNotFound,
+  IResponseSuccessJson,
+  ResponseErrorNotFound,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+
+import { RequiredParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_param";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseErrorQuery,
+  ResponseErrorQuery
+} from "io-functions-commons/dist/src/utils/response";
+
+import {
+  SERVICE_MODEL_ID_FIELD,
+  ServiceModel
+} from "io-functions-commons/dist/src/models/service";
+
+import { identity } from "fp-ts/lib/function";
+import { isSome } from "fp-ts/lib/Option";
+import { taskEither } from "fp-ts/lib/TaskEither";
+import { ServiceId } from "io-functions-commons/dist/generated/definitions/ServiceId";
+import { ServicePublic } from "io-functions-commons/dist/generated/definitions/ServicePublic";
+import { retrievedServiceToPublic } from "../utils/services";
+
+type IGetServiceByLatestRevisionHandlerRet =
+  | IResponseSuccessJson<ServicePublic>
+  | IResponseErrorNotFound
+  | IResponseErrorQuery;
+
+type IGetServiceByLatestRevisionHandler = (
+  serviceId: ServiceId
+) => Promise<IGetServiceByLatestRevisionHandlerRet>;
+
+const getServiceByLatestRevisionTask = (
+  serviceModel: ServiceModel,
+  serviceId: ServiceId
+) =>
+  serviceModel
+    .findOneByQuery({
```

Comment:
As docs:

> @deprecated use getQueryIterator + asyncIterableToArray

Reply:
Ok I think I get it. You are using `findOneByQuery` because you wanted to override the query.https://github.com/pagopa/io-functions-commons/blob/1a55096465b62b025a3111d02461d87f837411d9/src/utils/cosmosdb_model_versioned.ts#L180
That's because model id equality is case-sensitive, but how are we in a situation where this matters?
@AleDore can we come up with a more systemic solution?

Reply:
@balanza : In this case it matters because normally serviceIds are in upper case, but there is a service (namely `IO - L' App dei Servizi Pubblici`), that has been saved with a lowercase id. At the moment it's more complex to change the id's case instead of applying non case sensitive logic into retrieve queries :)
On the other side to put this case in a more systemic solution, maybe we have to implement something into io-functions-commons, don't we ?

Reply:
I think we should spend some time in the future to make this model more solid. I don't think patching the model would be fine, maybe we can solve the root problem. Why is the client looking for an ID which is not the same value as it's been stored in the db? Where that ID come from?